src/main/java/edu/ie3/datamodel/io/sink/SqlSink.java

```diff
@@ -0,0 +1,507 @@
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SqlSink {
+
+  protected static final Logger log = LoggerFactory.getLogger(SqlSink.class);
+
+  private final SqlConnector connector;
+  private final DatabaseNamingStrategy databaseNamingStrategy;
+  private final ProcessorProvider processorProvider;
+  private final String schemaName;
+
+  private static final String TIME_SERIES = "time_series";
+
+  public SqlSink(
+      String schemaName, DatabaseNamingStrategy databaseNamingStrategy, SqlConnector connector)
+      throws EntityProcessorException {
+    this(schemaName, new ProcessorProvider(), databaseNamingStrategy, connector);
+  }
+
+  public SqlSink(
+      String schemaName,
+      ProcessorProvider processorProvider,
+      DatabaseNamingStrategy databaseNamingStrategy,
+      SqlConnector connector) {
+    this.connector = connector;
+    this.databaseNamingStrategy = databaseNamingStrategy;
+    this.processorProvider = processorProvider;
+    this.schemaName = schemaName;
+  }
+
+  public void shutdown() {
+    connector.shutdown();
+  }
+
+  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+
+  /**
+   * Entry point of a data sink to persist multiple entities in a collection.
+   *
+   * @param entities a collection of entities that should be persisted
+   * @param identifier identifier of the grid
+   * @param <C> bounded to be all unique entities. Handling of specific entities is normally then
+   *     executed by a specific {@link EntityProcessor}
+   */
+  public <C extends Entity> void persistAll(Collection<C> entities, DbGridMetadata identifier) {
+    // Extract nested entities and add them to the set of entities
+    Set<C> entitiesToAdd = new HashSet<>(entities); // entities to persist
+    entities.forEach(
+        entity -> {
+          if (entity instanceof NestedEntity nestedEntity) {
+            try {
+              entitiesToAdd.addAll(
+                  (List<C>) Extractor.extractElements(nestedEntity).stream().toList());
+            } catch (ExtractorException e) {
+              log.error(
+                  String.format(
+                      "An error occurred during extraction of nested entity'%s': ",
+                      entity.getClass()),
+                  e);
+            }
+          }
+        });
+
+    // Persist the entities in hierarchic order to avoid failure because of foreign keys
+    for (Class<?> cls : hierarchicInsert()) {
+      persistMixedList(
+          entitiesToAdd.stream().filter(ent -> cls.isAssignableFrom(ent.getClass())).toList(),
+          identifier);
+      entitiesToAdd.removeIf(
+          ent ->
+              cls.isAssignableFrom(
+                  ent.getClass())); // maybe it's not necessary, but I'm not sure if there are
+      // entities who aren't in the hierarchic structure
```

Comment:
This is definitely neccessary if we call persistMixedList with the remaining entities afterwards, otherwise we would persist every elements infinitely if I understand correctly.